pandas/core/arrays/interval.py

```diff
@@ -562,6 +563,104 @@ def from_tuples(
 
         return cls.from_arrays(left, right, closed, copy=False, dtype=dtype)
 
+    _interval_shared_docs["from_strings"] = textwrap.dedent(
+        """
+        Construct from string representations of the left and right bounds.
+
+        Parameters
+        ----------
+        data : array-like (1-dimensional)
+            Strings representing the Interval's to parse.
+        copy : bool, default False
+            Copy the data.
+        dtype : dtype, optional
+            If None, dtype will be inferred.
+
+        Returns
+        -------
+        %(klass)s
+
+        Raises
+        ------
+        ValueError
+            When a string cannot be parsed as an Interval
+            When the dtype of the string cannot be parsed as either float,
+            Timestamp or Timedelta
+
+        See Also
+        --------
+        interval_range : Function to create a fixed frequency IntervalIndex.
+        %(klass)s.from_breaks : Construct an %(klass)s from an array of
+            splits.
+        %(klass)s.from_tuples : Construct an %(klass)s from an
+            array-like of tuples.
+
+        %(examples)s\
+        """
+    )
+
+    @classmethod
+    @Appender(
+        _interval_shared_docs["from_strings"]
+        % {
+            "klass": "IntervalIndex",
+            "examples": textwrap.dedent(
+                """\
+        Examples
+        --------
+        >>> pd.IntervalIndex.from_strings(["(0.0, 1.0]", "(1.0, 2.0]"])
+        IntervalIndex([(0.0, 1.0], (1.0, 2.0]],
+                       dtype='interval[float64, right]')
+        """
+            ),
+        }
+    )
+    def from_strings(
+        cls: type[IntervalArrayT],
+        data: Sequence[str],
+    ) -> IntervalArrayT:
+        # These need to be imported here to avoid circular dependencies.
+        from pandas.core.tools.datetimes import to_datetime
+        from pandas.core.tools.timedeltas import to_timedelta
+
+        intervals: list[Interval] = []
+        for string in data:
```

Comment:
why are you not using regexes?

this is very fixed to a single closed type - is there a reason?

Reply:
Changed to a stricter regex matching in e6e072a.

May I ask you to elaborate on your second question? I am afraid I don't understand!

Reply:
2nd question was this *only* supports closed intervals  (its ok if that is the case), but then pls open a followup issue. or ideally support all of the 'closed' types

Reply:
or 3rd option is to simply document this.

Reply:
@erikmannerfelt if you'd response here. basically this only supports left open-right closed. its fine, pls just document this format.

Reply:
Support for all closed types added in a8cc3b2caf5d3d93030eccf6a8c3759b804817ca!

One future room for improvement is dynamic parsing of the closed type (allowing the `closed` argument to be `None`). I opted to not do that for now, as it would require quite descriptive error messages if e.g. the first is closed on the right and the subsequent intervals are closed on the left.